### PR TITLE
switch to go 1.9

### PR DESCRIPTION
**Pull request by deads2k**
/assign skuznets

needed to land the rebase

### CODE REVIEW

**Comment by imcsk8** [comment]
LGTM

**Comment by deads2k** [comment]
@stevekuznetsov fail

**Comment by stevekuznetsov** [comment]
Fixed in https://github.com/openshift/aos-cd-jobs/pull/928

/test verify

**Comment by stevekuznetsov** [comment]
```
[INFO] Detected go version: go version go1.9 linux/amd64.
[FATAL] Please install Go version go1.9 or use PERMISSIVE_GO=y to bypass this check.
```

FUN.

**Comment by deads2k** [comment]
> [INFO] Detected go version: go version go1.9 linux/amd64.
> [FATAL] Please install Go version go1.9 or use PERMISSIVE_GO=y to bypass this check.
> FUN.

hacked the check out.

**Comment by deads2k** [comment]
/retest

I'll update unit tomorrow

**Comment by stevekuznetsov** [comment]
So here is our snafu:

 - we _still_ have not either containerized or gotten rid of the dependency on the Docker daemon for the `end-to-end` suite
 - that suite uses `go test` to run integration-type tests that touch the Docker daemon directly
 - that action requires `go` of the correct version on the host
 - upgrading `go` on the host is only possible (without crazy hacks) in the base AMI for all jobs for all repos

I think the best course of action today is:

 - merge the refactors in this PR (as long as they are compatible with 1.8)
 - move all CI to 1.9
 - merge the bash bits

File: hack/lib/util/golang.sh [hunk 1]
```diff
@@ -11,7 +11,8 @@ function os::golang::verify_go_version() {
 	if [[ "${go_version[2]}" != go1.8* ]]; then
```
Comment (soltysh) [inline hack/lib/util/golang.sh]: This should be `OS_REQUIRED_GO_VERSION`

Comment (deads2k) [reply hack/lib/util/golang.sh]: > This should be OS_REQUIRED_GO_VERSION

done

**Comment by deads2k** [comment]
> I think the best course of action today is:
> 
> merge the refactors in this PR (as long as they are compatible with 1.8)
> move all CI to 1.9
> merge the bash bits

I'll split the bits apart. We'll need to green button them in.

**Comment by liggitt** [comment]
> move all CI to 1.9

does that mean 3.6, 3.7, 3.8 cannot run CI? or do we have a base AMI per branch?

**Comment by stevekuznetsov** [comment]
Historically no, Origin does not test older versions. OSE does.

**Comment by stevekuznetsov** [comment]
Well, not quite -- the tests running inside of a container (all but registry and router integration) use the Go version declared in the repo under `hack/` and select the correct version regardless of the AMI.

**Comment by deads2k** [comment]
no longer required.  I think https://github.com/openshift/origin/issues/17755 is the dangling bit now.

**Pull request closed without merging**